### PR TITLE
doc: update kubernetes CNI requirements

### LINKED ISSUE #26531: Docs: Remove Deprecated network-plugin Kubelet Flag
From the [docs](https://docs.cilium.io/en/stable/network/kubernetes/requirements/):
>[CNI](https://docs.cilium.io/en/stable/glossary/#term-CNI) - Container Network Interface is the plugin layer used by Kubernetes to delegate networking configuration. CNI must be enabled in your Kubernetes cluster in order to install Cilium. This is done by passing --network-plugin=cni to kubelet on all nodes.

The `--network-plugin` flag was [removed](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) in k8s v1.24 and should be removed or updated accordinly in the docs.

### PR DESCRIPTION
**Pull request by swastik959**
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

I have updated the docs about enabling the cilium CNI 

Fixes: #26531

```release-note
docs: Update instructions for enabling the CNI in Kubernetes 1.24+
```

### CODE REVIEW

File: Documentation/network/kubernetes/requirements.rst [hunk 1]
```diff
@@ -37,10 +37,8 @@ Enable CNI in Kubernetes
...
-delegate networking configuration. CNI must be enabled in your Kubernetes
-cluster in order to install Cilium. This is done by passing
-``--network-plugin=cni`` to kubelet on all nodes. For more information, see
-the `Kubernetes CNI network-plugins documentation <https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/>`_.
+delegate networking configuration. Prior to Kubernetes 1.24, the CNI plugins could also be managed by the kubelet using the ``cni-bin-dir`` and ``network-plugin`` command-line parameters. These command-line parameters were removed in Kubernetes 1.24, with management of the CNI no longer in scope for kubelet.
+ For more information, see the `Kubernetes CNI network-plugins documentation <https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/>`_. To enable the cilium CNI plugin you need to configure the CNI plugin located at ``/etc/cni/net.d/`` . you can do this by making a config file for at the designated location . for making a config file you can refer to : `cilium config documentation <https://docs.cilium.io/en/stable/configuration/>`_.
```
Comment (qmonnet) [inline Documentation/network/kubernetes/requirements.rst]: A few things here:

- Let's keep present tense for how things work on K8s 1.24, given that Cilium still supports this version.
- It's not clear what “also” refers to. I think you copy-pasted this sentence from K8s's docs? Please let's adjust it to fit in the context.
- Mind the indentation (` For more information, ...`), the unexpected space at the beginning of the line breaks the build.
- Please fix the case and punctuation.
- Please wrap the lines, keep them under 80 characters in width or so.
- The internal reference (for configuration) needs to be fixed.

```suggestion
delegate networking configuration:

- Up to Kubernetes 1.23, you can enable the CNI in your cluster by passing ``--network-plugin=cni`` to Kubelet on all nodes.
- Starting from Kubernetes 1.24, this command-line parameter has been removed and kubelet no longer manages the CNI. For more information, see the `Kubernetes CNI network-plugins documentation`_. To enable the Cilium CNI plugin, configure the CNI plugin located at ``/etc/cni/net.d/``, by making a configuration file at the designated location. For more information about creating a configuration file, refer to :ref:`Cilium configuration <configuration>`.

.. _Kubernetes CNI network-plugins documentation: https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks a lot!

I see some things that can be improved, please comments and suggestions below. Please also add a commit log to explain why the update is necessary. Ideally, make your commit title more expressive, too.

File: Documentation/network/kubernetes/requirements.rst (hunk 1 above)
Comment (squeed) [inline Documentation/network/kubernetes/requirements.rst]: At this point, the document is no longer really listing a requirement and instead providing generic background information. In that spirit, we should document that Ciliium requires a CRI-compliant runtime that uses CNI for networking. In that vein, Cilium is tested to work with cri-o and containerd.

Comment (swastik959) [reply Documentation/network/kubernetes/requirements.rst]: @squeed sorry for being inactive I was busy in some chores . Do you mean to say that I should remove all of it and just put "Cilium requires a CRI-compliant runtime that utilizes CNI for networking. It has been tested to work with cri-o and containerd. Ensure that your container runtime is compatible with Cilium."

Comment (squeed) [reply Documentation/network/kubernetes/requirements.rst]: That would be my preference. This document's purpose is to capture Cilium's requirements, not identify Kubernetes upgrade concerns.

**Comment by lmb** [comment]
@swastik959 could you rebase, squash and address the feedback if you still want to land this PR?

**Comment by joestringer** [comment]
(marking draft while feedback is addressed. When you're ready to continue, click "Ready for review" at the bottom of the page)

**Pull request closed without merging**